packages/experimental-package/src/components/RemoveModal/RemovalModal.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { Modal, TextInput } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+
+export const RemovalModal = ({
+  body,
+  className,
+  header,
+  onRequestClose,
+  onRequestSubmit,
+  open,
+  primaryButtonText,
+  resource,
+  secondaryButtonText,
+  subheader,
+  textConfirmation,
+}) => {
+  const [userInput, setUserInput] = useState('');
+  const onChangeHandler = (e) => {
+    setUserInput(e.target.value);
+  };
+  const primaryButtonDisabled = textConfirmation && userInput !== resource;
+  return (
+    <Modal
+      danger
```

Comment:
The `iconDescription` prop of the modal needs to be set (from a client prop) to avoid the hardcoded, English 'Close' title for the close icon.